pkg/auth/dependency/interaction/provider_commit.go

```diff
@@ -99,11 +100,34 @@ func (p *Provider) onCommitSignup(i *Interaction, intent *IntentSignup) error {
 	return nil
 }
 
-func (p *Provider) onCommitAddIdentity(i *Interaction, intent *IntentAddIdentity) error {
+func (p *Provider) onCommitAddIdentity(i *Interaction, intent *IntentAddIdentity, userID string) error {
 	err := p.checkIdentitiesDuplicated(i.NewIdentities)
 	if err != nil {
 		return err
 	}
+
+	user, err := p.User.Get(userID)
+	if err != nil {
+		return err
+	}
+
+	for _, i := range i.NewIdentities {
+		identity := model.Identity{
+			Type:   string(i.Type),
+			Claims: i.Claims,
+		}
+		err = p.Hooks.DispatchEvent(
+			event.IdentityCreateEvent{
```

Comment:
Would it be better to dispatch a single event for adding multiple identities? The hook receiver may find it hard to correlate separate hook events that actually originate from a single interaction.

cc @kiootic

Reply:
I think it is not necessary. In most cases only one identity would be added at a time, and I don't think it would be very useful to allow such correlation.